pgjdbc/src/main/java/org/postgresql/core/Utils.java

```diff
@@ -129,10 +130,15 @@ public static StringBuilder escapeIdentifier(@Nullable StringBuilder sbuf, Strin
    * @param sbuf Either StringBuffer or StringBuilder as we do not expect any IOException to be
    *        thrown.
    * @param value value to append
+   * @param quoteIdentifier some ORM's send quote the identifier. If this is the case they can set
+   *                        the QUOTE_RETURNING_IDENTIFIER property to false
    */
-  private static void doAppendEscapedIdentifier(Appendable sbuf, String value) throws SQLException {
+  private static void doAppendEscapedIdentifier(Appendable sbuf, String value, boolean quoteIdentifier) throws SQLException {
+
     try {
-      sbuf.append('"');
+      if ( quoteIdentifier ) {
```

Comment:
This does not make sense. If `quoteIdentifier` is false (i.e. explicitly overridden via the new property) then it won't be wrapped in quotes but any internal quotes will be doubled. It should be doing nothing in that situation as the expectation is that the values provided are already ready to be used, either they're quoted or they do not require quoting.

I don't like adding more options to change method handling but if we're going to do this then it needs to be done higher up. Whatever is calling this escapeIdentifier(...) method should decide whether it is called at all. It shouldn't be another parameter to escapeIdentifier(...) that turns it into a no-op.

Reply:
Well now we are getting into the subtleties aren't we. I presume they don't actually quote internally, they just wrap every identifier in quotes. But perhaps I am presuming too much?
@jnehlmeier ?

Reply:
It should behave the same as with the SQL query. I guess you fully trust the caller to provide a correct SQL query, including correct quotes, right? If yes, this should be done with the column names for the returning clause as well. That would mean to revert changes in this method and not call the method at all in case the new connection parameter is used.

Reply:
I'm sure there will be someone who wants something different as this is implementation specific, but fair enough